pkg/http/server.go

```diff
@@ -81,6 +78,26 @@ func Run(cfg config.MutualPeersConfig) {
 	log.Info("Server Started...")
 	log.Info("Listening on port: " + httpPort)
 
+	// Check if the config has the consensusNode field defined to generate the metric from the Genesis Hash data.
+	if cfg.MutualPeers[0].ConsensusNode != "" {
+		// Initialise the goroutine to generate the metric in the background, only if we specify the node in the config.
+		go func() {
+			log.Info("Consensus node defined to get the first block")
+			for {
+				err := GenerateHashMetrics(cfg)
+				// check if err is nil, if so, that means that Torch was able to generate the metric.
+				if err == nil {
+					log.Info("Metric generated for the first block...")
+					// The metric was successfully generated, stop the retries.
+					break
+				}
+
+				// Wait for the retry interval before the next execution
+				time.Sleep(retryInterval)
+			}
+		}()
+	}
+
```

Comment:
It's better to separate this into dedicated functions
In addition they need to return errors and we need to handle them properly